### ISSUE TITLE
`cat FILE | gopass insert` produces unexpected results 

### ISSUE THREAD

**Issue by dim** [gopasspw/gopass#2104]
### Summary

I am aware that the correct way of inserting file contents is via `cat FILE |  gopass cat` but - currently - `gopass insert` can also accept values from `STDIN` which may have unexpected consequences. Please see example below:

### Steps To Reproduce

```shell
$ cat x.txt 
1
2
3
$ cat x.txt | gopass insert x 
$ gopass cat x
2
3
```

```shell
$ cat x.txt | gopass cat x 
$ gopass cat x
1
2
3
```
 
### Expected behavior

I would expect `gopass insert` to behave like `gopass cat` when accepting inputs from STDIN or explicitly reject STDIN with an error message.

### Environment

- OS: Ubuntu 20.04
- OS version: `Linux 5.11.0-41-generic #45~20.04.1-Ubuntu SMP Wed Nov 10 10:20:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`
- gopass Version: `gopass 1.13.0+c574e920b532606894b3e64db059f3e3a838b0c9 (c574e920b532606894b3e64db059f3e3a838b0c9) go1.17.3 linux amd64`
- Installation method: gopass release binary

**Comment by AnomalRoil** [comment]
This seems to be an issue with `cat` specifically, indeed.

What does `gopass show -f -n x` display in the first case? (`-f` forces password output and `-n` disables parsing completely to be sure nothing strange is going on, but both shouldn't be required in normal usage, it's just to test here.)

My guess is that `gopass cat` is somehow discarding the password itself and only displaying the body.

**Comment by dim** [comment]
I don't think it's an issue with `cat` as it is just printing the binary value. The output is also correct when the value was inserted with `cat`.

```shell
$ cat x.txt 
1
2
3
$ cat x.txt | gopass insert x 
$ gopass show -f -n x
Secret: x

1
2
3
```

**Comment by AnomalRoil** [comment]
FYI 
```
echo """1
2
3""" | gopass cat c1`
```

Results in a file containing:
```
                                                                                                                                                                            
content-disposition: attachment; filename="STDIN"
content-transfer-encoding: Base64
MQoyCjMK
```
Notice the empty first line, that's where we assume the password is located?

Now, 
echo """
1
2
3""" | gopass insert i1
```
Works with `cat`, this really seems like `cat` is just ignoring the `password` line is there is one and only parsing what we're usually calling "body".

**Comment by AnomalRoil** [comment]
We can test "displaying just the password" using `gopass show -o x`:

in the case where it's "filled" using insert, it will display `1` but in the case where it's filled with `cat` it will display `Error: empty secret`

**Comment by AnomalRoil** [comment]
Found the culprit:
https://github.com/gopasspw/gopass/blob/0387c552ea48b843b5af1462f56c8afc39a88a80/internal/action/binary.go#L230-L232

@dominikschulz  What's your take, should we instead "default" to a `show -f -n` when using `cat` and it is not base64 encoded data after decryption?

**Comment by dominikschulz** [comment]
Thanks for digging.
That's a bug. I have probably mixed up `sec.Body()` with `sec.Bytes()` and never tested it this way.

Fixing it is easy, but we should likely have a test case for it, too.

**Comment by dim** [comment]
Thanks for the quick fix, much appreciated! I would also really appreciate a v1.13.1 too :)

**Comment by dominikschulz** [comment]
If it's easy to cherry-pick this fix onto the v1.13.0 then we can do a v1.13.1, yes. Otherwise we'll have to wait for the Go 1.18 release.

**Comment by AnomalRoil** [comment]
Shall we try and fix `mv` in #2079 as well for 1.13.1 or not?

**Comment by dominikschulz** [comment]
As long as `we` doesn't include myself: yes 😄 

Seriously: I'm not sure I will have time to look into that, but I'd love to get it fixed.
